src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -0,0 +1,3123 @@
+            writer.WriteLine("{");
+            writer.Write($"    public static Regex Instance {{ get; }} = ");
+
+            // If we can't support custom generation for this regex, spit out a Regex constructor call.
+            if (!SupportsCustomCodeGeneration(rm))
+            {
+                writer.WriteLine($"new Regex({patternExpression}, {optionsExpression}, {timeoutExpression});");
+                writer.WriteLine("}");
+                return;
+            }
+
+            writer.WriteLine($"new {id}();");
+            writer.WriteLine();
+            writer.WriteLine($"    private {id}()");
+            writer.WriteLine($"    {{");
+            writer.WriteLine($"        pattern = {patternExpression};");
+            writer.WriteLine($"        roptions = {optionsExpression};");
+            writer.WriteLine($"        internalMatchTimeout = {timeoutExpression};");
+            writer.WriteLine($"        factory = new RunnerFactory();");
+            if (rm.Code.Caps is not null)
+            {
+                writer.Write($"        Caps = new Hashtable {{");
+                AppendHashtableContents(writer, rm.Code.Caps);
+                writer.WriteLine(" };");
+            }
+            if (rm.Tree.CapNames is not null)
+            {
+                writer.Write($"        CapNames = new Hashtable {{");
+                AppendHashtableContents(writer, rm.Tree.CapNames);
+                writer.WriteLine(" };");
+            }
+            if (rm.Tree.CapsList is not null)
+            {
+                writer.Write($"        capslist = new string[] {{");
+                string separator = "";
+                foreach (string s in rm.Tree.CapsList)
+                {
+                    writer.Write(separator);
+                    writer.Write(Literal(s));
+                    separator = ", ";
+                }
+                writer.WriteLine(" };");
+            }
+            writer.WriteLine($"        capsize = {rm.Code.CapSize};");
+            writer.WriteLine($"        InitializeReferences();");
+            writer.WriteLine($"    }}");
+            writer.WriteLine("    ");
+            writer.WriteLine($"    private sealed class RunnerFactory : RegexRunnerFactory");
+            writer.WriteLine($"    {{");
+            writer.WriteLine($"        protected override RegexRunner CreateInstance() => new Runner();");
+            writer.WriteLine();
+            writer.WriteLine($"        private sealed class Runner : RegexRunner");
+            writer.WriteLine($"        {{");
+
+            // Main implementation methods
+            writer.WriteLine($"            protected override void InitTrackCount() => runtrackcount = {rm.Code.TrackCount};");
+            writer.WriteLine();
+            writer.WriteLine($"            protected override bool FindFirstChar()");
+            writer.WriteLine($"            {{");
+            writer.Indent += 4;
```

Comment:
> Is it because we are already indented 3 "manually"?

Yeah.

For most of the code I rely on the indent from the writer, but here, it felt cleaner to just manually indent all the class boilerplate, to make it a bit easier to read in the generator.